ideascube/configuration/management/commands/config.py

```diff
@@ -0,0 +1,153 @@
+    get_all_namespaces, get_config_data, get_namespaced_configs,
+    )
+
+
+class Command(BaseCommand):
+    help = 'Manage server configuration'
+
+    def add_arguments(self, parser):
+        subs = parser.add_subparsers(
+            title='Commands', dest='cmd', metavar='',
+            parser_class=argparse.ArgumentParser)
+
+        get = subs.add_parser(
+            'get', help='Get the current value of a configuration option')
+        get.add_argument('namespace', help='The configuration namespace')
+        get.add_argument('key', help='The configuration key')
+        get.set_defaults(func=self.get_config)
+
+        help = subs.add_parser(
+            'help', help='Print help about a configuration option')
+        help.add_argument('namespace', help='The configuration namespace')
+        help.add_argument('key', help='The configuration key')
+        help.set_defaults(func=self.help_for_config)
+
+        list = subs.add_parser(
+            'list', help='List configuration namespaces and keys')
+        list.add_argument(
+            'namespace', nargs='?',
+            help='Only list configuration keys for this namespace')
+        list.set_defaults(func=self.list_configs)
+
+        reset = subs.add_parser(
+            'reset', help='Reset a configuration option to its default value')
+        reset.add_argument('namespace', help='The configuration namespace')
+        reset.add_argument('key', help='The configuration key')
+        reset.set_defaults(func=self.reset_config)
+
+        set = subs.add_parser(
+            'set', help='Set the value of a configuration option')
+        set.add_argument('namespace', help='The configuration namespace')
+        set.add_argument('key', help='The configuration key')
+        set.add_argument('value', help='The new value')
+        set.set_defaults(func=self.set_config)
+
+        self.parser = parser
+
+    def handle(self, *_, **options):
+        if 'func' not in options:
+            self.parser.print_help()
+            self.parser.exit(1)
+
+        options['func'](options)
+
+    def _evaluate_value(self, value):
+        if value in ('true', 'false'):
+            # Special case booleans for a better UX
+            value = value.capitalize()
+
+        try:
+            return ast.literal_eval(value)
+
+        except (SyntaxError, ValueError):
+            # Admin probably wanted a string and didn't use enough quotes
+            # examples:
+            #     $ ideascube config set server site-name thename
+            #     $ ideascube config set server site-name 'the name'
+            return str(value)
+
+    def get_config(self, options):
+        namespace = options['namespace']
+        key = options['key']
+
+        try:
+            value = get_config(namespace, key)
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print('%r' % value)
```

Comment:
Maybe find a way to tell the user if the config is not set and we are returning the default value ?

I not sure of how to do it.
A simple print could break (or complicate) using script.
Just print en stderr ?
Add another command or option to get_config ?